cxx/pclean/pclean.cc

```diff
@@ -70,77 +70,55 @@ int main(int argc, char** argv) {
     std::cout << "Error reading schema file" << schema_fn << "\n";
   }
 
-  // Translate schema
-  std::cout << "Making schema helper ...\n";
-  PCleanSchemaHelper schema_helper(
+  // Make GenDB
+  std::cout << "Making GenDB model ...\n";
+  GenDB gendb(
+      &prng,
       pclean_schema,
       result["only_final_emissions"].as<bool>(),
       result["record_class_is_clean"].as<bool>());
-  std::cout << "Translating schema ...\n";
-  std::map<std::string, std::vector<std::string>> annotated_domains_for_relations;
-  T_schema hirm_schema = schema_helper.make_hirm_schema(
-      &annotated_domains_for_relations);
 
   // Read observations
   std::cout << "Reading observations ...\n";
   std::string obs_fn = result["obs"].as<std::string>();
   std::cout << "Reading observations file from " << obs_fn << "\n";
   DataFrame df = DataFrame::from_csv(obs_fn);
 
-  // Create model
-  std::cout << "Creating hirm ...\n";
-  HIRM hirm(hirm_schema, &prng);
-
   // Incorporate observations.
-  std::cout << "Translating observations ...\n";
-  T_observations observations = translate_observations(
-      df, hirm_schema, annotated_domains_for_relations);
-
-  std::string heldout_fn = result["heldout"].as<std::string>();
-  T_observations heldout_obs;
-  T_observations encoding_observations;
-  if (heldout_fn.empty()) {
-    encoding_observations = observations;
-  } else {
-    std::cout << "Loading held out observations from " << heldout_fn << std::endl;
-    DataFrame heldout_df = DataFrame::from_csv(heldout_fn);
-    heldout_obs = translate_observations(
-        heldout_df, hirm_schema, annotated_domains_for_relations);
-    encoding_observations = merge_observations(observations, heldout_obs);
-  }
-
-  std::cout << "Encoding observations ...\n";
-  T_encoding encoding = calculate_encoding(hirm_schema, encoding_observations);
-
   std::cout << "Incorporating observations ...\n";
-  incorporate_observations(&prng, &hirm, encoding, observations);
+  incorporate_observations(&prng, &gendb, df);
 
   // Run inference
   std::cout << "Running inference ...\n";
-  inference_hirm(&prng, &hirm,
```

Comment:
Can we still run HIRM with a PClean-like schema, i.e. Model 6? I think that's something we should try to keep (maybe with a bool flag for HIRM vs full GenDB, and calling incorporate_observations, inference_hirm etc on gendb.hirm)

Reply:
So not only can we do this, it is currently the only thing we can do -- all inference.cc::inference_gendb does right now is directly call inference_hirm.

But I agree that as we add the ability to inference_gendb to transition entity assignments, we should add flags to keep Model 5 & 6 behavior.

Reply:
Sounds good. Could you add those, as Model 7 inference comes together?

Reply:
Sure.